### PR TITLE
Handle exceptions in 5.2, Closes lucadegasperi/oauth2-server-laravel#641

### PR DESCRIPTION
**Pull request by mstephens**
Hi guys

This fixes the changes made in Laravel 5.2.7 onwards regarding the way in which middleware is handled via the routing pipeline.

Cheers

### CODE REVIEW

**Comment by mrgswift** [comment]
@mstephens Great! Thanks!

**Comment by mstephens** [comment]
PS. If you're now using Middleware groups, make sure this Middleware is the first in your list :]

File: src/Middleware/OAuthExceptionHandlerMiddleware.php [hunk 1]
```diff
@@ -33,14 +32,20 @@ class OAuthExceptionHandlerMiddleware
...
+
+            return $response;
         } catch (OAuthException $e) {
             $data = [
                 'error' => $e->errorType,
                 'error_description' => $e->getMessage(),
             ];
 
-            return new JsonResponse($data, $e->httpStatusCode, $e->getHttpHeaders());
+            return response()->json($data, $e->httpStatusCode, $e->getHttpHeaders());
```
Comment (vinkla) [inline src/Middleware/OAuthExceptionHandlerMiddleware.php]: Please use the class instead of the helper method.

Comment (mstephens) [inline src/Middleware/OAuthExceptionHandlerMiddleware.php]: @vinkla reverted back :+1:

**Comment by cesar-garay** [comment]
That works fine for me, Thanks so much!!

**Comment by vinkla** [comment]
@mstephens could you please squash your commits?

**Comment by mstephens** [comment]
@vinkla, sure there you go

**Comment by vinkla** [comment]
@mstephens Thanks!

**Comment by JuanDMeGon** [comment]
Hello guys. Is this pull request will merged in the repository? It seems to solve the issue. Thanks!

**Comment by vinkla** [comment]
Thanks @mstephens!

**Comment by JuanDMeGon** [comment]
Thank you guys! Nice job @mstephens.
Best wishes :+1:

**Comment by aninditkarmakar** [comment]
When is this commit going to appear in a release? I desperately need this fix in my application.

**Comment by vinkla** [comment]
It is added in 5.1.4

**Comment by mstephens** [comment]
```
"lucadegasperi/oauth2-server-laravel": "5.1.*",
```

Of your composer.json file

**Comment by oflannabhra** [comment]
I'm using v `5.1.4` with Lumen 5.2 and am still having some issues with `grant_type => password`. 

[This section](https://github.com/lucadegasperi/oauth2-server-laravel/blob/master/docs/authorization-server/password.md) of the documentation suggests that the `callback` should use `Auth::once()` to check if a request has the proper credentials. 

With the changes to `Authorization` in Laravel / Lumen, `Auth::validate()` requires access to a `$request`, however, the `callback` does not receive a `Request $request` variable. Anyone able to give some direction here?

#654

**Pull request merged**